### PR TITLE
BB-435 (test): Allow creation of multiple test entities

### PR DESCRIPTION
**Pull request by MonkeyDo**
### Problem
Errors are thrown in various places in create-entities test helpers when trying to create multiple test entities because of duplicate IDs.

### Solution
Instead of a static id, create a random ID every time where needed.

Also cleaned up unused function;

### Areas of Impact
test/test-helpers/create-entities.js

### CODE REVIEW

**Comment by MonkeyDo** [comment]
@prabalsingh24 I finally got up to making a PR for this code.
Care to have a look and try it out, see if it works for you?

**Comment by prabalsingh24** [comment]
> @prabalsingh24 I finally got up to making a PR for this code.
> Care to have a look and try it out, see if it works for you?

Wow, using random number. Very clever solution I must say!  Thanks. I will have a look at this.

File: test/test-helpers/create-entities.js [hunk 1]
```diff
@@ -242,15 +261,14 @@ export async function createWork(optionalBBID) {
...
 	const languageSetId = await createLanguageSet();
 
 	const workAttribs = {
 		bbid,
 		languageSetId,
-		typeId: setData.id
+		typeId: random.number()
 	};
-	await new WorkType({...setData, label: 'Work Type 1'})
+	await new WorkType({id: workAttribs.typeId, label: 'Work Type 1'})
```
Comment (prabalsingh24) [inline test/test-helpers/create-entities.js]: `label` has to be unique too. Maybe replace this with ```  label: `Work Type ${workAttribs.typeId}` ```

File: test/test-helpers/create-entities.js [hunk 2]
```diff
@@ -274,23 +292,24 @@ export async function createEditionGroup(optionalBBID) {
...
-		genderId: setData.id,
-		typeId: setData.id
+		genderId: editorAttribs.genderId,
+		typeId: random.number()
 	};
-	await new Area({...setData, gid: uuidv4(), name: 'Rlyeh'})
+	await new Area({gid: uuidv4(), id: areaId, name: 'Rlyeh'})
 		.save(null, {method: 'insert'});
-	await new AuthorType({...setData, label: 'Author Type 1'})
+	await new AuthorType({id: authorAttribs.typeId, label: 'Author Type 1'})
```
Comment (prabalsingh24) [inline test/test-helpers/create-entities.js]: Here also ``` label: `Author Type ${authorAttribs.typeId}` ```

File: test/test-helpers/create-entities.js [hunk 3]
```diff
@@ -311,11 +330,11 @@ export async function createPublisher(optionalBBID) {
...
 		endYear: 2012,
 		ended: true,
-		typeId: setData.id
+		typeId: random.number()
 	};
-	await new Area({...setData, gid: uuidv4(), name: 'Rlyeh'})
+	await new Area({gid: uuidv4(), id: publisherAttribs.areaId, name: 'Rlyeh'})
 		.save(null, {method: 'insert'});
-	await new PublisherType({...setData, label: 'Publisher Type 1'})
+	await new PublisherType({id: publisherAttribs.typeId, label: 'Publisher Type 1'})
```
Comment (prabalsingh24) [inline test/test-helpers/create-entities.js]: ``` label: `Publisher Type ${publisherAttribs.typeId} ```

File: test/test-helpers/create-entities.js [hunk 4]
```diff
@@ -261,9 +279,9 @@ export async function createEditionGroup(optionalBBID) {
 	await createEntityPrerequisites();
 	const editionGroupAttribs = {
 		bbid,
-		typeId: setData.id
+		typeId: random.number()
 	};
-	await new EditionGroupType({...setData, label: 'Edition Group Type 1'})
+	await new EditionGroupType({id: editionGroupAttribs.typeId, label: 'Edition Group Type 1'})
```
Comment (prabalsingh24) [inline test/test-helpers/create-entities.js]: Replace `label` with ``` label: `Edition Group Type ${editionGroupAttribs.typeId}` ```

File: test/test-helpers/create-entities.js [hunk 5]
```diff
@@ -242,15 +261,14 @@ export async function createWork(optionalBBID) {
...
 	const workAttribs = {
 		bbid,
 		languageSetId,
-		typeId: setData.id
+		typeId: random.number()
 	};
-	await new WorkType({...setData, label: 'Work Type 1'})
+	await new WorkType({id: workAttribs.typeId, label: `Work Type ${workAttribs.typeId}`})
 		.save(null, {method: 'insert'});
 	await new Work({...entityAttribs, ...workAttribs})
```
Comment (prabalsingh24) [inline test/test-helpers/create-entities.js]:
```
return new Work({...entityAttribs, ...workAttribs})
		.save(null, {method: 'insert'}); 
```

Comment (MonkeyDo) [reply test/test-helpers/create-entities.js]: I don't think that's a bad idea at all, and could be useful i some cases.

**Comment by MonkeyDo** [comment]
@prabalsingh24
Thanks for the suggestiong.
I want to point out that in some cases, it'll be better to fetch the newly created entity to get its data rather that use the entity returned by createWork, createAuthor, etc.
That way you ensure that what you expect is in the database.
There are cases however where we might need to compare attributes between the created entity and the one fetched from the database after creation, so it's useful to return it.

Anything else you think might be useful or missing?

**Comment by prabalsingh24** [comment]
Yeah thanks :D
No this is it I guess. 
I am able to make changes I want using this. I should be able to write tests without any problem now (hopefully) xD

**Pull request merged**